File: test/nodejs/duckpgq_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`duckpgq extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.DUCKPGQ_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
    });

    it('duckpgq function should return expected string', function (done) {
        db.all("SELECT duckpgq('Sam') as value;", function (err, res) {
            if (err) throw err;
            assert.deepEqual(res, [{value: "Duckpgq Sam 🐥"}]);
            done();
        });
    });

    it('duckpgq_openssl_version function should return expected string', function (done) {
        db.all("SELECT duckpgq_openssl_version('Michael') as value;", function (err, res) {
            if (err) throw err;
            assert(res[0].value.startsWith('Duckpgq Michael, my linked OpenSSL version is OpenSSL'));
            done();
        });
    });
});

File: test/python/duckpgq_test.py
import duckdb
import os
import pytest


# Get a fresh connection to DuckDB with the duckpgq extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('DUCKPGQ_EXTENSION_BINARY_PATH')
    if extension_binary == '':
        raise Exception('Please make sure the `DUCKPGQ_EXTENSION_BINARY_PATH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    return conn


def test_duckpgq(duckdb_conn):
    duckdb_conn.execute("SELECT duckpgq('Sam') as value;")
    res = duckdb_conn.fetchall()
    assert (res[0][0] == "Duckpgq Sam 🐥");


def test_property_graph(duckdb_conn):
    duckdb_conn.execute("CREATE TABLE foo(i bigint)")
    duckdb_conn.execute("INSERT INTO foo(i) VALUES (1)")
    duckdb_conn.execute("-CREATE PROPERTY GRAPH t VERTEX TABLES (foo);")
    duckdb_conn.execute("-FROM GRAPH_TABLE(t MATCH (f:foo))")
    res = duckdb_conn.fetchall()
    assert (res[0][0] == 1)
